packages/core/src/App/Containers/AccountSignupModal/account-signup-modal.jsx

```diff
@@ -54,20 +55,28 @@ const validateSignup = (values, residence_list) => {
     return errors;
 };
 
-const AccountSignup = ({ enableApp, isModalVisible, clients_country, onSignup, residence_list }) => {
+const AccountSignup = ({
+    enableApp,
+    isModalVisible,
+    clients_country,
+    onSignup,
+    residence_list,
+    isEuCountrySelected,
+}) => {
     const { is_dashboard } = React.useContext(PlatformContext);
     const [api_error, setApiError] = React.useState(false);
     const [is_loading, setIsLoading] = React.useState(true);
     const [country, setCountry] = React.useState('');
     const [pw_input, setPWInput] = React.useState('');
-    const [has_valid_residence, setHasValidResidence] = React.useState(false);
+    const [selected_residence, setSelectedResidence] = React.useState('');
+    const [is_eu_resident, setIsEuResident] = React.useState(false);
 
     const updatePassword = new_password => {
         setPWInput(new_password);
     };
 
-    const onResidenceSelection = () => {
-        setHasValidResidence(true);
+    const onResidenceSelection = values => {
+        setSelectedResidence(values);
```

Comment:
you can rely on `values instead of `selected_residence` to setIsEuResident. In this case you can remove useEffect below